Canister label system UI

Artorp opened the pull request:
1) Updated the nanomap
2) Added new ease-of-use UI for the new canister labeling system. Here's how it looks:

![new ui](https://cloud.githubusercontent.com/assets/147366/6157869/2c53da54-b245-11e4-9250-58564785f0a0.png)

https://github.com/ParadiseSS13/Paradise/issues/292

~~Preferably wait with merging till I get time to take a closer look at PJB3005's concerns~~

Done.

PJB3005 commented:
lookes nice!

PJB3005 commented on code/game/machinery/atmoalter/canister.dm:
is this hardcoding needed?

PJB3005 commented on code/game/machinery/atmoalter/canister.dm:
I think we need a safety check here, because I could easily make canisters next to invisible with template modification

PJB3005 commented on code/game/machinery/atmoalter/canister.dm:
I think we need a safety check here, because I could easily make canisters next to invisible with template modification

PJB3005 commented on code/game/machinery/atmoalter/canister.dm:
is this hardcoding needed, or is that a consequence of nanoUI

Artorp commented on code/game/machinery/atmoalter/canister.dm:
You're right, I'll have to take a closer look at this later and add a safety check.

Artorp commented on code/game/machinery/atmoalter/canister.dm:
not really needed, I've since implemented a for loop to make the code scalable

Artorp commented:
Ready to merge.

Changes:

1) Fixed canister flags, removed duplicate calls of update_icon, sanitization of name.
2) Safety check added for both colors and decals, to prevent possible template modifications (thanks, @PJB3005)
3) Added scalability. The code should properly adjust to future additions of colors and decals, as long as the .dmi is changed appropriately. The template file would need pixel height adjustments to keep it pretty, but functionality wise it too should support more items.
